pypsa-to-iamc

```diff
@@ -0,0 +1,634 @@
+                var[sh][v.replace('Fuel consumption','Efficiency')]=ct*var[sheets['Yearly_generation_supply']][v.replace('Fuel consumption','Generation|Yearly')]\
+                                                                        /var[sheets['fuel_consumption_supply']][v]
+            
+             #Efficiency : hydro (reservoir, ror)
+
+            var[sh]['Efficiency|Electricity|Hydro|river'] = n.generators.efficiency.filter(like ='ror').filter(like =country).mean()
+            var[sh]['Efficiency|Electricity|Hydro|dam'] = n.storage_units.efficiency_dispatch.filter(like ='hydro').filter(like =country).mean()
+            var[sh]['Efficiency|Electricity|Hydro'] = var[sh]['Efficiency|Electricity|Hydro|dam']   #currently equal in the model since eff(dam)=eff(river)
+             
+                
+             #Efficiency : storage (PHS, battery, H2 storage)   
+            
+            var[sh]['Efficiency|Flexibility|Electricity Storage|Medium duration'] = (
+                  (n.storage_units.efficiency_dispatch.filter(like ='PHS').filter(like =country).mean()*n.storage_units.efficiency_store.filter(like ='PHS').filter(like =country).mean()+
+                  n.links.efficiency.filter(like ='H2 Electrolysis').filter(like =country).mean()*n.links.efficiency.filter(like ='H2 Fuel Cell').filter(like =country).mean())/2)
+            
+            var[sh]['Efficiency|Flexibility|Electricity Storage|Short duration'] = (n.links.efficiency.filter(like ='battery charger').filter(like =country).mean()
+              *n.links.efficiency.filter(like ='battery discharger').filter(like =country).mean())
+            
+            var[sh]['Efficiency|Flexibility|Electricity Storage'] =  (var[sh]['Efficiency|Flexibility|Electricity Storage|Medium duration']+
+            var[sh]['Efficiency|Flexibility|Electricity Storage|Short duration'])/2
+               
+
+             #Efficiency :Electricity (Resistive heater, heat pump)
+           
+            var[sh]['Efficiency|Heat|Electricity|Direct'] = (var[sheets['Yearly_generation_supply']]['Generation|Yearly|Heat|Electricity|Direct']
+             /(0.001*ct*n.links_t.p0.filter(like ='resistive heater').filter(like =country).sum().sum()))
+            var[sh]['Efficiency|Heat|Electricity|Heat pump'] = (var[sheets['Yearly_generation_supply']]['Generation|Yearly|Heat|Electricity|Heat pump']
+             /(0.001*ct*n.links_t.p0.filter(like ='heat pump').filter(like =country).sum().sum()))
+            var[sh]['Efficiency|Heat|Electricity']=((var[sheets['Yearly_generation_supply']]['Generation|Yearly|Heat|Electricity|Direct']+var[sheets['Yearly_generation_supply']]['Generation|Yearly|Heat|Electricity|Heat pump'])
+             /(0.001*ct*(n.links_t.p0.filter(like ='resistive heater').filter(like =country).sum().sum()+n.links_t.p0.filter(like ='heat pump').filter(like =country).sum().sum())))
```

Comment:
these should be efficiencies, not p0

for resistive heater efficiency is constant so no need to do the average

Reply:
1)This is using the same (efficiency = energy generation (sum of p1) / consumption (sum of p0) ). Is it incorrect?
2) for resistive heaters, I saw different numbers in the model (like 0.9 for "residential rural" and 0.99 for "urban central").

Reply:
1) Can I ask what is the reason for not using efficiency (and keep (sum of p1)/(sum of p0).
The former should be preferred since it is more clear for future readers. 
2) Ok, then, lets report an average of "individual electric boilers" and "central electric boilers"